src/java.desktop/unix/native/libawt_xawt/awt/gtk3_interface.c

```diff
@@ -2861,17 +2863,45 @@ static void transform_detail_string (const gchar *detail,
     }
 }
 
+inline static int scale_down_to_plus_inf(int what, int scale) {
+    return (int)ceilf(what / (float)scale);
+}
+
+inline static int scale_down_to_minus_inf(int what, int scale) {
+    return (int)floorf(what / (float)scale);
+}
+
 static gboolean gtk3_get_drawable_data(JNIEnv *env, jintArray pixelArray,
      int x, jint y, jint width, jint height, jint jwidth, int dx, int dy,
                                                                    jint scale) {
```

Comment:
Is this scale parameter is simply ignored? If the passed parameters are always in the device space and the array is allocated properly then we should not care about this scale(especially in case of gtk2)?

Reply:
Yes please, this "scale" just make the code complicated w/o reason.

Reply:
That's done; please, take a look at the updated code.